### PR TITLE
Hive account creation

### PR DESCRIPTION
**Pull request by sisygoboom**
- Create hive account
- tests
- refactor hive repository into hive chain repo and (linked) hive account repo
- fixed broken mock auth guard

### CODE REVIEW

File: src/services/api/dto/RequestHiveAccount.dto.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty } from 'class-validator';
+
+export class RequestHiveAccountDto {
+  @IsNotEmpty()
+  @ApiProperty({
+    description: 'Did of the account',
+    default: 'test-did',
```
Comment (Geo25rey) [inline src/services/api/dto/RequestHiveAccount.dto.ts:8]:
```suggestion
    description: 'Username for the new Hive account',
    default: 'myUsername123',
```

File: src/services/auth/auth.controller.ts [hunk 2]
```diff
@@ -382,17 +385,6 @@ export class AuthController {
...
-    schema: {
-      properties: {
-        username: {
-          type: 'string',
-          default: 'test-account',
-          description: 'Username of requested HIVE account',
-        },
-      },
-    },
-  })
```
Comment (Geo25rey) [inline src/services/auth/auth.controller.ts:395]: What's wrong with this?

File: src/services/hive/hive.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  readonly #hiveRepository: HiveChainRepository;
+  readonly #hiveAccountRepository: HiveAccountRepository;
+  readonly #logger: LoggerService = new Logger(HiveService.name);
+
+  constructor(hiveAccountRepository: HiveAccountRepository, hiveRepository: HiveChainRepository) {
+    this.#hiveRepository = hiveRepository;
+    this.#hiveAccountRepository = hiveAccountRepository;
+  }
+
+  async requestHiveAccount(hiveUsername: string, sub: string) {
```
Comment (Geo25rey) [inline src/services/hive/hive.service.ts:17]: This isn't used?

**Comment by Geo25rey** [comment]
Where is the "Vote on Hive using linked account"? Is this change implicit?

File: src/services/auth/auth.controller.ts (hunk 2 above)
Comment (sisygoboom) [reply src/services/auth/auth.controller.ts:395]: Its already done in the dto; this overwrites it

Comment (Geo25rey) [reply src/services/auth/auth.controller.ts:395]: Sounds good to me

File: src/services/hive/hive.service.ts (hunk 3 above)
Comment (sisygoboom) [reply src/services/hive/hive.service.ts:17]: Used in the auth controller

**Comment by sisygoboom** [comment]
Sorry, I was working on two features at the same time as there were dependencies shared in the refactor, and it made sense to merge first. This is hive account creation, not voting.

File: src/services/auth/auth.controller.ts [hunk 4]
```diff
@@ -382,17 +385,6 @@ export class AuthController {
...
   @ApiOkResponse({
     description: 'Account created',
     schema: {
```
Comment (Geo25rey) [inline src/services/auth/auth.controller.ts:390]: The return type of the function doesn't match this schema

File: src/services/hive/hive.service.ts (hunk 3 above)
Comment (Geo25rey) [reply src/services/hive/hive.service.ts:17]: Got it

**Comment by Geo25rey** [comment]
Np all good. Just check to be sure I didn't miss something is all.

**Review by Geo25rey — APPROVED** [review]
LGTM

**Comment by Geo25rey** [comment]
Probably should fix failing test

**Pull request merged**